lib/src/localization.dart

```diff
@@ -0,0 +1,106 @@
+import 'dart:convert';
+import 'dart:ui';
+
+import 'package:easy_localization/src/translations.dart';
+import 'package:intl/intl.dart';
+
+import 'asset_loader.dart';
+
+class Localization {
+  Translations _translations;
+  Locale _locale;
+
+  String path;
+  bool useOnlyLangCode;
+  final RegExp _replaceArgRegex = RegExp(r'{}');
+
+  Localization._();
+
+  static Localization _instance;
+  static Localization get instance =>
+      _instance ?? (_instance = Localization._());
+
+  static Future<bool> load(
+    Locale locale, {
+    String path,
+    bool useOnlyLangCode,
+    AssetLoader assetLoader,
+  }) async {
+    assert(locale != null &&
+        path != null &&
+        useOnlyLangCode != null &&
+        assetLoader != null);
+    instance._locale = locale;
+    instance.path = path;
+    instance.useOnlyLangCode = useOnlyLangCode;
+
+    String localePath = instance.getLocalePath();
+    String data = await assetLoader.load(localePath);
```

Comment:
Maybe we can change it a little so assetLoader returns `Map<String, dynamic>` instead of a String. This way, if the user wants to do something special during JSON.decode, such as using [json_annotation](https://pub.dev/packages/json_annotation), he can do it.

Reply:
I think that the best way to move forward 👍